### PR TITLE
aws - waf - tag support

### PR DESCRIPTION
**Pull request by JohnHillegass**
Adds tag support for WAF and Regional WAF

### CODE REVIEW

File: tests/test_schema.py [hunk 1]
```diff
@@ -224,24 +224,18 @@ def test_semantic_mode_error(self):
...
-                             {'or': [{'tag:Owner': 'absent'},
-                                     {'tag:ResponsibleParty': 'absent'},
-                                     {'tag:Contact': 'absent'},
-                                     {'tag:Creator': 'absent'}]}],
-                 'name': 'tagging-compliance-waf',
+                {'actions': [{'key': 'AES3000',
+                              'type': 'encryption',
+                              'value': 'This resource should have AES3000 encryption'}],
+                 'description': 'Identify resources which lack our outrageous cipher',
+                 'name': 'bogus-policy',
```
Comment (JohnHillegass) [inline tests/test_schema.py:231]: This test seems to just be here to test that it errors, so I made it completely outrageous as I am sure that is what people used to think about being able to tag WAFs 😆

**Comment by kapilt** [comment]
lgtm, but can you confirm using in practice, there's a possible oddity wrt to resource group tagging and which version of waf.

**Comment by JohnHillegass** [comment]
@kapilt, great question! It works for WAFv2 and Regional WAF but not WAFv1. I'll need to add some more to have this reflect correctly (WAFv2 resource etc)

**Comment by kapilt** [comment]
@JohnHillegass at the moment this is particular resource is about waf v1 regional support, so if thats confirmed working, its good to merge.

**Comment by JohnHillegass** [comment]
Sounds good! I just need to rerecord some test data and will push back up. Will add a note for me to start messing with getting this working for v1 and v2 global later.

**Comment by JohnHillegass** [comment]
@kapilt, good to go now! I had to augment it to get it working but made sure not to shadow anything in the resource or source augments

**Comment by kapilt** [comment]
hmm.. independent of this pr, just noticed there's an issue with the resource group tag stuff works and the test recording changes in #5561 as changing the account id, from the default thats in tests/common.py will affect arn generation for resource group tag augmentation, which will cause a mismatch on tag fetch. it only affects resources where we need to generate the arn (ie arn is not an attribute).

**Comment by JohnHillegass** [comment]
Oh man, total side effect! Instead of doing the 123456789123 replace we can just set everything to that test account value. Sound good ?

**Comment by kapilt** [comment]
> Oh man, total side effect! Instead of doing the 123456789123 replace we can just set everything to that test account value. Sound good ?

sounds good, thanks for the follow up pr.

**Comment by kapilt** [comment]
hmm... we're picking up waf-regional tags in the waf-enabled/set-waf filter/actions with this, not sure that we want that since the tags augments are extraneous for those operations. we'd have to parameterize query resource manager resources with an augment boolean, and skip the tag fetch.

**Comment by JohnHillegass** [comment]
Thats a good point! I hadn't considered how extraneous it is. I am not really sure how to proceed with the parameterization so I'll reach out in chat. My first attempt was to scan the policy filters and actions to see if the word tag is included and then set the Tags to an empty list if it isn't, though that is kludgey at best.

**Comment by JohnHillegass** [comment]
To resolve this, I added a boolean to query resource manager resources that will augment by default but can explicitly be set to don't augment/False/

File: c7n/resources/appelb.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ def process(self, resources, event=None):
         name_id_map = {}
         resource_map = {}
 
-        wafs = self.manager.get_resource_manager('waf-regional').resources()
+        wafs = self.manager.get_resource_manager('waf-regional').resources(None, False)
```
Comment (kapilt) [inline c7n/resources/appelb.py:204]: augment=False instead of passing None, False

File: c7n/resources/appelb.py [hunk 3]
```diff
@@ -266,7 +266,7 @@ def validate(self):
         return self
 
     def process(self, resources):
-        wafs = self.manager.get_resource_manager('waf-regional').resources()
+        wafs = self.manager.get_resource_manager('waf-regional').resources(None, False)
```
Comment (kapilt) [inline c7n/resources/appelb.py:269]: same here, augment=False

File: c7n/resources/appelb.py (hunk 2 above)
Comment (JohnHillegass) [reply c7n/resources/appelb.py]: Ah thanks, that reads much better. Resolved in f9d6ed0.

**Pull request merged**